indexing-service/src/main/java/io/druid/indexing/common/task/IndexTask.java

```diff
@@ -898,6 +899,7 @@ public boolean isAppendToExisting()
     private static final boolean DEFAULT_GUARANTEE_ROLLUP = false;
     private static final boolean DEFAULT_REPORT_PARSE_EXCEPTIONS = false;
     private static final long DEFAULT_PUBLISH_TIMEOUT = 0;
+    private static final Period DEFAULT_LOCK_TIMEOUT = new Period("PT5m");
```

Comment:
@akashdw I agree on that the default lock timeout should be smaller than http client timeout for now, so 5 min is the default value. However, I think the lock timeout configuration should be a separated one because 1) the problem still exist if users override the lock timeout with some smaller value than http client timeout, and 2) the fundamental problem looks that any messages aren't sent while waiting for a lock and should be fixed by sending kind of heartbeat messages in the future. If this problem is fixed, the lock timeout can be longer than the http client timeout.

I'll note this problem in the doc. Does it make sense?